### PR TITLE
Add rs/moq-bench: a relay load-generator CLI

### PR DESCRIPTION
**Pull request by kixelated**
## Summary

Adds `rs/moq-bench`, a new binary (`moq-bench`) for benchmarking a remote MoQ server by opening many QUIC connections and driving synthetic media through them.

Every knob is a `[min, max]` range that is **rolled once per connection**, so a single config can describe a heterogeneous swarm (some connections at 24fps, others at 60, etc).

For a run, `moq-bench` establishes **A** connections. Each connection:
- publishes **B** broadcasts, each with a single track;
- subscribes to **C** other broadcasts discovered via announcements;
- produces **D** frames/sec per track, each **E** bytes large;
- splits frames into groups of **F** frames.

The first frame of every group is a JSON keyframe describing the rolled parameters (connection id, broadcast path, group sequence, fps, frame size, group size, wall-clock timestamp). The remaining **F** frames are zeroed. **F may be 0**, in which case each group is a lone keyframe, useful for stressing the announce/subscribe control plane rather than the data path.

To avoid a thundering herd, connections and subscriptions are staggered over a `--startup` ramp window instead of all firing at once.

## Stats: received vs skipped groups

The subscribe side reports delivery, not just throughput. Subscribers read groups via `recv_group` (arrival order, out-of-order included) and track each subscription's sequence span; a span wider than the count received means groups in between were skipped. The periodic report gains `recv_groups`, `lost_groups`, and a `loss` percentage. The newest group of each subscription is the live frontier and is excluded from the count, since groups just behind it may still be in flight; a gap is only blamed once a higher group confirms it was skipped. The first keyframe of each subscription is parsed back into the publisher's shape (fps/frame size/group size), so a subscriber works against peers it didn't publish.

Loss here means dropped/skipped groups, not QUIC packet loss (which the transport already repairs). Latency is intentionally left out: groups carry only a per-group timestamp, so the meaningful metric would be jitter rather than one-way latency. That can come later, optionally behind a `hang` container with per-frame timestamps.

## Config

- CLI flags always win over the TOML file, mirroring `moq-relay`'s parse → merge → re-parse so an absent flag never clobbers a TOML value (every overridable field is `Option<T>`).
- Each range accepts a scalar (`--fps 30`), a `min:max` string (`--fps 24:60`), or a TOML table (`fps = { min = 24, max = 60 }`).
- Client TLS/QUIC flags come from `moq-native`, identical to `moq-cli`/`moq-relay`.
- Presets under `config/`: `hd.toml` (HD video 24-60fps), `sd.toml` (SD, more viewers), `audio.toml` (Opus-like), `announce.toml` (control-plane stress, `group_size = 0`).

```bash
moq-bench --file rs/moq-bench/config/hd.toml --url https://relay.example.com --connections 500
```

## Test plan

- [x] `cargo test -p moq-bench` — range parsing/sampling, CLI-overrides-TOML merge + defaults, two in-process media-loop tests (keyframe JSON + zeroed payload; the `group_size = 0` lone-keyframe edge case), and gap-tracker math (sequence holes and out-of-order arrival) using paused time.
- [x] `cargo clippy -p moq-bench --all-targets` clean; `cargo fmt` clean.
- [x] Smoke-ran the binary: it rolls per-connection params, staggers connection startup across the ramp, logs throughput stats, and shuts down cleanly on `--duration` / Ctrl-C.
- [x] End-to-end against a live relay (`cdn.moq.wtf`): a 4-connection publish→announce→subscribe mesh fully converges (12/12 subscriptions, steady recv_fps) at **0.00% group loss**, and the JSON keyframe shape round-trips over the wire (subscriber recovers fps/frame_size/group_size from peers).

## Cross-package sync

Added `moq-bench` to the crate index in `CLAUDE.md`. No existing wire/public APIs changed (new self-contained crate), so no paired packages needed updates. Targeting `main` as a self-contained additive change; happy to retarget `dev` if preferred.

(Written by Claude)

---
_Generated by [Claude Code](https://claude.ai/code/session_01SPMx2iNsBh8LNi4LN1VKMv)_

### CODE REVIEW

File: rs/moq-bench/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+use connection::Rolled;
+use rand::RngExt;
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+	// TODO: It would be nice to remove this and rely on feature flags only.
+	// However, some dependency is pulling in `ring` and I don't know why, so meh for now.
+	rustls::crypto::aws_lc_rs::default_provider()
+		.install_default()
+		.expect("failed to install default crypto provider");
```
[the remark this answers is not in the document]
Comment (kixelated) [reply rs/moq-bench/src/main.rs:22]: Keeping `.expect()` here on purpose: `install_default()` returns `Result<(), Arc<CryptoProvider>>`, and `Arc<CryptoProvider>` doesn't implement `std::error::Error`, so `.context(...)?` won't compile (anyhow's `Context` is only implemented for `Result<T, E: std::error::Error + ...>`). This also matches the existing `moq-cli` and `moq-relay` `main`s, which install the provider the same way. The other four review comments have been addressed in the latest push.

(Written by Claude)

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (kixelated) [reply rs/moq-bench/src/main.rs:22]: Keeping `.expect` here: `install_default()` returns `Result<(), Arc<CryptoProvider>>`, and `Arc<CryptoProvider>` is not an `Error`, so `.context(...)?` would not compile. This matches `moq-cli`/`moq-relay`. (Written by Claude)
